backend/langate/network/models.py

```diff
@@ -97,3 +84,27 @@ def create_user_device(user: User, ip, name=None):
 
         device = UserDevice.objects.create(mac=mac, name=name, user=user, ip=ip, area=area)
         return device
+
+def generate_dev_name():
+    """
+        Generate a random device name based on a list of names
+    """
+    try:
+        with open("assets/misc/device_names.txt", "r", encoding="utf-8") as f:
+            lines = f.read().splitlines()
+            devices_count = Device.objects.count() + UserDevice.objects.count()
+            
+            if devices_count < len(lines) and devices_count < 1000 :
+                taken_names = Device.objects.values_list("name", flat=True)
+                taken_names.extend(UserDevice.objects.values_list("name", flat=True))
+                n = random.choice(lines)
+                while n in taken_names:
+                    n = random.choice(lines)
+                taken_names.append(n)
```

Comment:
taken_names is a local variable (even if python variable's scopes are strange), adding a value at the end of this function is useless.